src/artm/core/processor.cc

```diff
@@ -41,6 +41,77 @@ const float kProcessorEps = 1e-16;
 namespace artm {
 namespace core {
 
+static void CreateThetaCacheEntry(std::shared_ptr<DataLoaderCacheEntry> new_cache_entry_ptr,
+                                  DenseMatrix<float>* theta_matrix,
+                                  const Batch& batch,
+                                  const ModelName& model_name,
+                                  std::shared_ptr<ModelIncrement> model_increment,
+                                  int topic_size) {
+  if (new_cache_entry_ptr != nullptr) {
+    new_cache_entry_ptr->set_batch_uuid(batch.id());
+    new_cache_entry_ptr->set_model_name(model_name);
+    new_cache_entry_ptr->mutable_topic_name()->CopyFrom(model_increment->topic_model().topic_name());
+    for (int item_index = 0; item_index < batch.item_size(); ++item_index) {
+      const Item& item = batch.item(item_index);
+      new_cache_entry_ptr->add_item_id(item.id());
+      new_cache_entry_ptr->add_item_title(item.has_title() ? item.title() : std::string());
+      FloatArray* cached_theta = new_cache_entry_ptr->add_theta();
+      for (int topic_index = 0; topic_index < topic_size; ++topic_index) {
+        cached_theta->add_value((*theta_matrix)(topic_index, item_index));
+      }
+    }
+  }
+}
+
+static void CreatePtdwCacheEntry(std::shared_ptr<DataLoaderCacheEntry> new_cache_entry_ptr,
+                                 DenseMatrix<float>* ptdw_matrix,
+                                 const std::string& batch_uuid,
+                                 const ModelName& model_name,
+                                 std::shared_ptr<ModelIncrement> model_increment,
+                                 int item_id,
+                                 int topic_size) {
+  if (new_cache_entry_ptr != nullptr) {
+    new_cache_entry_ptr->set_batch_uuid(batch_uuid);
```

Comment:
I would rather move the following lines out from this method, and keep them in the ``ThreadFunction``.
```
new_cache_entry_ptr->set_batch_uuid(batch_uuid);
new_cache_entry_ptr->set_model_name(model_name);
new_cache_entry_ptr->mutable_topic_name()->CopyFrom(model_increment->topic_model().topic_name());
```
This would reduce the number of input parameters for ``CreatePtdwCacheEntry`` (and same for ``CreateThetaCacheEntry``. Also, copying ``topic_name`` for all items might be an overhead.